### PR TITLE
Added Vial support for CherryBStudio87 RGB TKL PCB

### PR DESCRIPTION
**Pull request by WrinkleKrinkles**
Added Vial support for CherryBStudio87 RGB TKL PCB. 

Static RGB LED is still functional (saturation, hue, and brightness control), however RGB preset animations were removed to preserve EEPROM space.

List of removed features:

1. **All of the in-built RGB animations were removed** (build the firmware again with several choices of RGB animations if you want to enable them)
2. Tap dance, space cadet, combo, ~~key override, mousekey, extrakey~~, and grave escape were disabled.
3. Dynamic keycap layer count were set into 2. 

Total firmware size after build: ~~23810/28672 (83%, 4862 bytes free)~~  27264/28672 (95%, 1408 bytes free) with reenabled mousekeys and extrakeys after reviews on the disabled RGB animations and keymap layer count.

Possible issues:

1. ~~In the VIAL GUI, the keyboard is still listing about 4 layers even with the maximum layer set into 2; haven't tested the displayed layer 2, 3, and 4 on their functionality~~
2. Possible issues on F13 stop working and the left arrow key stopped working when using WKL 7U layout (based on the issues propping up on the CherryBStudio's Discord channel)

**Possible security issue:**
Built and tested the firmware using the newest .json file that fixes the issue (per 22th May, 2022). The said .json file needed to be loaded in the "Design" tab of VIA in "Load Draft Definition" 

I was not able to verify the .json file on other points. In terms of functionality, it is currently working without any issue being reported. 

To do:

- [x] Testing other available layers on Vial GUI.
Fixed; was putting the wrong code in config.h and rules.mk
- [x] Repairing the folders (the build failed because the files were meant to be moved to **cb87rgb** instead of **cb87**)
- [x] Adding additional Readme files for a "quick guide" to enable animation
Simply change the #undef to #define on the list of RGB animations
- [x] Following-up for the latest fix on .json file of the board
At the present moment, no fix had been uploaded to the support channels. 

Further testing might be required. So far in personal use and using the F13 button with other keycodes still functional. Will update if the "permanent" fix for the .json files were available (if required). 

Only tested on ANSI, F13 WK, 6.25u space layout.

### CODE REVIEW

File: keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
+TAP_DANCE_ENABLE = no
+COMBO_ENABLE = no
+KEY_OVERRIDE_ENABLE = no
+MOUSEKEY_ENABLE = no
+EXTRAKEY_ENABLE = no
+SPACE_CADET_ENABLE = no
+GRAVE_ESC_ENABLE = no 
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (xyzz) [inline keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk]: this is supposed to go into config.h

**Review by xyzz — CHANGES_REQUESTED** [review]

File: keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk (hunk 1 above)
Comment (WrinkleKrinkles) [reply keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk]: Thanks for the heads-up. Reuploading changes on the disabled RGB lights too. 

There was a typo on the #undef function, saved a little bit of memory size.

**Review by WrinkleKrinkles — COMMENTED** [review]
Revised keymap layer count and fixed typo on disabling RGB animations.

Reenabled mousekeys and extrakeys (sufficient memory size after alterations).

**Review by WrinkleKrinkles — COMMENTED** [review]
Revised the config.h, keymap.c, and rules.mk files.

File: keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk [hunk 2]
```diff
@@ -4,8 +4,5 @@ VIAL_ENABLE = yes
 TAP_DANCE_ENABLE = no
 COMBO_ENABLE = no
 KEY_OVERRIDE_ENABLE = no
-MOUSEKEY_ENABLE = no
-EXTRAKEY_ENABLE = no
 SPACE_CADET_ENABLE = no
 GRAVE_ESC_ENABLE = no 
-#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (WrinkleKrinkles) [inline keyboards/cherrybstudio/cb87/keymaps/vial/rules.mk]: Moved to config.h as requested

**Review by WrinkleKrinkles — COMMENTED** [review]
Moved to the appropriate folder.

Build failed because I had uploaded cb87rgb files to cb87 files.

**Pull request merged**